### PR TITLE
Jormungandr: fix stay-in that pass midnight

### PR DESCRIPTION
**Pull request by TeXitoi**

### CODE REVIEW

File: source/routing/raptor_visitors.h [hunk 1]
```diff
@@ -42,6 +42,15 @@ struct raptor_visitor {
        return vj->next_vj;
     }
 
+    inline DateTime
+    get_base_dt_extension(DateTime base_dt, const type::VehicleJourney* vj) const {
+        if (vj->prev_vj->stop_time_list.back().departure_time > vj->stop_time_list.front().departure_time) {
+            std::cout << "+1d" << std::endl;
```
Comment (antoine-de) [inline source/routing/raptor_visitors.h]: debug line

File: source/routing/raptor_visitors.h [hunk 2]
```diff
@@ -42,6 +42,15 @@ struct raptor_visitor {
        return vj->next_vj;
     }
 
+    inline DateTime
+    get_base_dt_extension(DateTime base_dt, const type::VehicleJourney* vj) const {
+        if (vj->prev_vj->stop_time_list.back().departure_time > vj->stop_time_list.front().departure_time) {
```
Comment (antoine-de) [inline source/routing/raptor_visitors.h:47]: prev_vj cannot be null there ?

File: source/routing/raptor_visitors.h [hunk 3]
```diff
@@ -94,6 +103,15 @@ struct raptor_reverse_visitor {
        return vj->prev_vj;
     }
 
+    inline DateTime
+    get_base_dt_extension(DateTime base_dt, const type::VehicleJourney* vj) const {
+        if (vj->next_vj->stop_time_list.front().departure_time < vj->stop_time_list.back().departure_time) {
+            std::cout << "-1d" << std::endl;
```
Comment (antoine-de) [inline source/routing/raptor_visitors.h]: debug line

File: source/routing/tests/raptor_test.cpp [hunk 4]
```diff
@@ -3238,3 +3238,37 @@ BOOST_AUTO_TEST_CASE(reverse_pass_midnight_with_boardings){
...
+    vj2->next_vj = nullptr;
+    RAPTOR raptor(*b.data);
+
+    auto result = raptor.compute(
+        b.data->pt_data->stop_areas_map["S1"], b.data->pt_data->stop_areas_map["S5"],
+        "23:00"_t, 1, DateTimeUtils::inf, type::RTLevel::Base, 2_min
+    );
+
+    BOOST_REQUIRE_EQUAL(result.size(), 1);
+    result.at(0).print();
```
Comment (antoine-de) [inline source/routing/tests/raptor_test.cpp]: debug line

File: source/routing/tests/raptor_test.cpp [hunk 5]
```diff
@@ -3238,3 +3238,37 @@ BOOST_AUTO_TEST_CASE(reverse_pass_midnight_with_boardings){
     result = compute(5);
     BOOST_REQUIRE_EQUAL(result.size(), 0);
 }
+
+BOOST_AUTO_TEST_CASE(stay_in_pass_midnight){
```
Comment (antoine-de) [inline source/routing/tests/raptor_test.cpp]: can you add a global test comment (even if it's quite clear with the name and dataset)

File: source/routing/raptor_visitors.h (hunk 2 above)
Comment (TeXitoi) [reply source/routing/raptor_visitors.h:47]: As you must provide a base_dt from the previous vj, it can't be null.

**Review by antoine-de — APPROVED** [review]

**Pull request merged**